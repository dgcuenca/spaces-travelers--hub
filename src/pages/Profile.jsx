import React from 'react';
import { useSelector } from 'react-redux';
import { reservedRockets } from '../features/rockets/rockets.slice';
import { getJoinedMissions } from '../features/missions/missions.slice';
import styles from './Profile.module.scss';

const Profile = () => {
  const rockets = useSelector(reservedRockets);
  const missions = useSelector(getJoinedMissions);
  return (
    <div className={styles.profile}>
      <div className={styles.profile__wrapper}>
        <h2 className={styles.profile__title}>My Missions</h2>
        <ul className={styles.profile__items}>
          {missions ? (
            missions.map((mission) => (
              <li
                className={styles.profile__item}
                key={mission.mission_id}
              >
                {mission.mission_name}
              </li>
            ))
          ) : (
            <li className={styles['profile__item--none']}>
              Sorry, currently no reserve mission to show...
            </li>
          )}
        </ul>
      </div>
      <div className={styles.profile__wrapper}>
        <h2 className={styles.profile__title}>My Rockets</h2>
        <ul className={styles.profile__items}>
          {rockets.length > 0 ? (
            rockets.map((rocket) => (
              <li
                className={styles.profile__item}
                key={rocket.id}
              >
                {rocket.name}
              </li>
            ))
          ) : (
            <li className={styles['profile__item--none']}>
              Sorry, currently no reserve rocket to show...
            </li>
          )}
        </ul>
      </div>
    </div>
  );
};

export default Profile;
