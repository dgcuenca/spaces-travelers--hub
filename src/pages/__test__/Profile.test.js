/**
 * @jest-environment jsdom
 */

import Profile from '../Profile';
import '@testing-library/jest-dom';
import renderWithProviders from '../../../testingHelper';

describe('Profile Page', () => {
  test('renders Profile Page', () => {
    const tree = renderWithProviders(<Profile />);
    expect(tree.toJSON).toMatchSnapshot();
  });
});
