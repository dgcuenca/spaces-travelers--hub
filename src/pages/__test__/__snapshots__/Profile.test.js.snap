// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page renders Profile Page 1`] = `undefined`;
